specs/AdaptiveCardsXamarinSpec.md

```diff
@@ -0,0 +1,231 @@
+
+## Dependencies
+This project depends entirely on the Android and iOS renderer and their API's. This project is a Xamarin bindings library so the only new code to be included are the xamarin transformations for the underlying renderers. For generating the bindings, Xamarin requires the generated binaries (aar file for Android and framework file for iOS) to be included.
+
+### What is a bindings library
+
+A bindings library is an assembly that wraps the native library (Objective-C/Java code) with C# wrappers so platform specific code can be invoked via C# calls. Xamarin.Android implements bindings by using Managed Callable Wrappers (MCW) which are a JNI bridge that are used when managed code needs to invoke Java code. In the case of iOS, to bind a library an API definition file is needed; this is merely a C# source file that contains C# interfaces that have been annotated with a handful of attributes that help drive the binding. The API definition is merely the contract that will be used to generate the API.
+
+## Architecture
+
+As can be seen in the diagram below, the Android and iOS bindings libraries make use our existing iOS/Android renderer libraries that already contain projected code from the shared C++ object model. The bindings also make use of the native APIs for both platforms yielding a C# library for every platform that can be consumed by any Xamarin platform.
+
+![image](assets/XamarinArchitecture.png)
+
+Something to consider is that Native Xamarin is designed to share the logic and services inside an application but the UI is independent for every platform which allows us to have three different object models.  
+
+## Feature Priority
+
+The priorities among the APIs to be ported into Xamarin for the pre-release version were decided by selecting which APIs would be the bare minimum for a consumer of the Xamarin package to be able to render a card would be P0 as well as any APIs that were part of a feature (such as featureRegistration). 
+
+For P1, the tasks were decided using the same features asked for P1 in the [Xamarin.Forms github issue](https://github.com/microsoft/AdaptiveCards/issues/3255) which states that custom rendering for actions and elements would be P1. 
+
+Finally, P2 would be any task that enhances the developer experience but are not required for using Adaptive Cards and P3 would be any APIs that may not be used by most consumers or are specific to only one platform.
+
+## Rendering
+
+### UWP
+
+For Xamarin rendering UWP doesn't require any changes as the rendering and API's are done in C#. The current renderer we provide must only be referenced as is or included as a nuget package.
+
+### Android
```

Comment:
It'd be super helpful to have some sample code in this spec that shows calls to some of these projections.